optuna/integration/botorch.py

```diff
@@ -609,11 +658,32 @@ def sample_relative(
         if self._candidates_func is None:
             self._candidates_func = _get_default_candidates_func(n_objectives=n_objectives)
 
+        completed_values = values[:n_completed_trials]
+        completed_params = params[:n_completed_trials]
+        if self._consider_running_trials:
+            running_params = params[n_completed_trials:]
+            running_params = running_params[~torch.isnan(running_params).any(dim=1)]
+        else:
+            running_params = None
+
         with manual_seed(self._seed):
             # `manual_seed` makes the default candidates functions reproducible.
             # `SobolQMCNormalSampler`'s constructor has a `seed` argument, but its behavior is
             # deterministic when the BoTorch's seed is fixed.
-            candidates = self._candidates_func(params, values, con, bounds)
+            try:
+                candidates = self._candidates_func(
+                    completed_params, completed_values, con, bounds, running_params
+                )
+            except TypeError:
+                if self._consider_running_trials:
+                    warnings.warn(
+                        f"The provided candidates_func {self._candidates_func.__name__} must take "
+                        "an argument pending_x to support option consider_running_trials=True. "
+                        "consider_running_trials option will be ignored."
+                    )
```

Comment:
I am not sure which of `warnings.warn(...)` or `_logger.warning(...)` is the preferable way of warning here. Please let me know if you prefer one over the other.